lib/hammer_cli_foreman_ansible/command_extensions/inventory.rb

```diff
@@ -0,0 +1,116 @@
+          collection :foreman_interfaces, _('Network interfaces') do
+            field :name, _('Interface Name')
+            field :identifier, _('Identifier')
+            field :attrs, _('Attributes'), Fields::Field, hide_blank: true
+            field :mac, _('MAC address')
+            field :ip, _('IPv4 address'), Fields::Field, hide_blank: true
+            field :ip6, _('IPv6 address'), Fields::Field, hide_blank: true
+            field :fqdn, _('FQDN')
+            field :virtual, _('Virtual')
+            field :link, _('Link')
+            field :managed, _('Managed')
+            field :primary, _('Primary')
+            field :provision, _('Provision')
+            field :subnet6, _('Subnet6')
+            field :tag, _('Tag')
+            field :attached_to, _('Attached to')
+            field :type, _('Type')
+            field :attached_devices, _('Attached devices')
+            collection :subnet, _('Subnet'), numbered: false, hide_blank: true do
+              field :name, _('Name')
+              field :network, _('Network')
+              field :mask, _('Mask')
+              field :gateway, _('Gateway')
+              field :dns_primary, _('Primary Dns')
+              field :dns_secondary, _('Secondary dns')
+              field :from, _('From')
+              field :to, _('To')
+              field :boot_mode, _('Boot mode')
+              field :ipam, _('IPAM')
+              field :vlanid, _('Vlanid')
+              field :mtu, _('MTU')
+              field :nic_delay, _('Nic delay')
+              field :network_type, _('Network type')
+              field :description, _('Description')
+            end
+          end
+
+          collection :foreman_users, _('Foreman Users'), numbered: false do
+            field :firstname, _('First name')
+            field :lastname, _('Last name')
+            field :mail, _('Email')
+            field :description, _('Description')
+            field :fullname, _('Full name')
+            field :ssh_authorized_keys, _('SSH authorized keys')
+          end
+
+          collection :foreman_ansible_roles, _('Foreman Ansible Roles'), numbered: false do
+            field :name, _("Role")
+          end
+        end
+      end
+
+      before_print do |data|
+        new_data = data['all']['hosts'].each_with_object([]) do |hostname, arr|
+          data['_meta']['hostvars'][hostname]['foreman']['foreman_users'] = data['_meta']['hostvars'][hostname]['foreman']['foreman_users']&.map { |u| u[1] }
+          data['_meta']['hostvars'][hostname]['foreman_ansible_roles'] =  data['_meta']['hostvars'][hostname]['foreman_ansible_roles']&.map{ |r| {"name" => r}}
+          data['_meta']['hostvars'][hostname].map {|label| data['_meta']['hostvars'][hostname]['foreman'][label[0]] = label[1]}
+          arr << {
+            'hostname' => hostname
+          }.merge(data['_meta']['hostvars'][hostname]['foreman']).merge
+        end
+        data.clear
+        data['results'] = new_data
+      end
```

Comment:
This should be a bit adjusted in case if we want to show exact response from the server without any changes:

```ruby
  before_print do |data, cmd_obj|
    unless cmd_obj.option_as_json?
      # ...
    end
  end
```
This way we explicitly check if we need to adjust the response since `before_print` will modify it.